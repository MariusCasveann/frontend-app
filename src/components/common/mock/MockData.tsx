import { GraphQLError } from 'graphql';
import { QueryAllProducts_allProducts } from './../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryAllDomains_allDomains } from './../../dashboard/atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { LightApplicationsOfProduct_lightApplicationsOfProduct } from './../../applications/graphql/__generated__/LightApplicationsOfProduct';
import { QueryBilling_billingQuery } from '../../dashboard/molecules/common/qraphql/__generated__/QueryBilling';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from '../../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { SelectItemProps } from '../molecules/SearchableSelect/SearchableSelectPresenter';

export const mockGraphQLError: GraphQLError = {
    message: 'FATALITY',
    name: 'error',
    locations: undefined,
    path: undefined,
    nodes: undefined,
    positions: undefined,
    source: undefined,
    originalError: undefined,
    extensions: undefined
};

export const mockSelectedDomain: QueryAllDomains_allDomains = {
    id: 102,
    name: 'Engineering Platform',
    __typename: 'Domain'
};

export const mockSelectedProduct: QueryAllProducts_allProducts = {
    id: 50051,
    logo: null,
    name: 'a',
    __typename: 'LightProductDTO'
};

export const mockedApplications: LightApplicationsOfProduct_lightApplicationsOfProduct[] = [
    {
        id: 50051,
        staticPictureId: null,
        name: 'a',
        service: false,
        __typename: 'LightApplicationDTO'
    }
];

export const mockFunction = jest.fn();

export const mockDataEmpty: QueryBilling_billingQuery[] = [];

// tslint:disable-next-line:no-any
export const createMockedState = (loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const userProductMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const preSelectedProduct: QueryAllProducts_allProducts = {
    __typename: 'LightProductDTO',
    id: 'ALL',
    logo: 'ALL',
    name: 'All Products'
};

export const currentUserMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: preSelectedProduct
        }
    }
});

export const selectedProductMockedState = (
    loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>
): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const buildDataExpectedEmptyResults = {
    labels: [],
    datasets: [
        {
            data: []
        }
    ]
};

export const mockErrorMessage = 'Error message';

export const mockTeamMember: TeamByProductId_teamByProductId_teamMembers = {
    __typename: 'User',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    lastName: 'Test',
    id: '11',
    roles: [
        {
            name: 'WRITE_ALL',
            __typename: 'Role'
        },
        {
            name: 'a-admin',
            __typename: 'Role'
        }
    ]
};

export const mockTeamByProductId: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: 2355,
    productOwner: {
        id: 11,
        firstName: 'TestUser',
        lastName: 'Test',
        __typename: 'User'
    },
    teamMembers: [
        {
            __typename: 'User',
            email: 'test1@mediamarktsaturn.com',
            firstName: 'Test',
            id: '1234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                }
            ]
        },
        {
            __typename: 'User',
            email: 'test2@mediamarktsaturn.com',
            firstName: 'Test2',
            id: '2234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                },
                {
                    name: 'test-admin',
                    __typename: 'Role'
                }
            ]
        }
    ]
};

export const mockCurrentUser: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: [
        {
            concretePermissions: [],
            id: 1300,
            name: 'DP God',
            readOnly: true,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 4651,
            name: 'READ_ALL',
            readOnly: false,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 69002,
            name: 'a-read',
            readOnly: true,
            __typename: 'Role'
        }
    ],
    staticPictureId: null,
    subscribedToNewsletter: true
};

export const mockEmptyTeam: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: null,
    productOwner: null,
    teamMembers: []
};

export const mockCurrenUserRole: QueryCurrentUser_currentUser_roles = {
    concretePermissions: [],
    id: 1300,
    name: 'DP God',
    readOnly: true,
    __typename: 'Role'
};

export const mockForm = {
    getFieldDecorator: () => () => false,
    getFieldError: mockFunction,
    getFieldsError: mockFunction,
    getFieldsValue: mockFunction,
    getFieldValue: mockFunction,
    isFieldValidating: mockFunction,
    setFields: mockFunction,
    setFieldsValue: mockFunction,
    validateFields: mockFunction,
    validateFieldsAndScroll: mockFunction,
    isFieldsTouched: mockFunction,
    isFieldTouched: mockFunction,
    resetFields: mockFunction
};

export const mockHistory = {
    length: 7,
    push: mockFunction,
    action: 'PUSH',
    location: {
        hash: '',
        pathname: '/path',
        search: '',
        state: {}
    },
    replace: mockFunction,
    go: mockFunction,
    goBack: mockFunction,
    goForward: mockFunction,
    block: mockFunction,
    createHref: mockFunction,
    listen: mockFunction
};

export const mockSelectData: SelectItemProps[] = [
    {
        id: '351952',
        label: 'Test test'
    },
    {
        id: '80503',
        label: 'Test1 test1'
    },
    {
        id: '412605',
        label: 'Test2 test2'
    }
];
