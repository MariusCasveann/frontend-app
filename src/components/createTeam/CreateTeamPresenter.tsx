import React, { FormEvent } from 'react';
import { Button, Col, Form, Row } from 'antd';

import { Card } from '../common/molecules/CardLayout/CardLayout';

import './team.css';
import SearchableSelectPresenter, {
    SelectItemProps
} from '../common/molecules/SearchableSelect/SearchableSelectPresenter';
import { buttonItemLayout, formLayout, getTeamFormRules } from './helpers/teamHelper';
import { WrappedFormUtils } from 'antd/lib/form/Form';
import { TeamByProductId_teamByProductId } from '../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';

const rules = getTeamFormRules();

interface CreateTeamPresenterProps {
    handleSubmit: (e: FormEvent<HTMLFormElement>) => void;
    teamMembersOptions: SelectItemProps[];
    productOwnerOptions: SelectItemProps[];
    setSearchProductOwnerInput: (value: string) => void;
    setSearchTeamMembersInput: (value: string) => void;
    form: WrappedFormUtils;
    loadingTeamMembers: boolean;
    loadingProductOwner: boolean;
    team: TeamByProductId_teamByProductId | null;
}

export default (props: CreateTeamPresenterProps) => {
    const {
        handleSubmit,
        teamMembersOptions,
        productOwnerOptions,
        setSearchProductOwnerInput,
        setSearchTeamMembersInput,
        form: { getFieldDecorator },
        loadingTeamMembers,
        loadingProductOwner,
        team
    } = props;

    return (
        <Card width={100} title={team ? 'Update Team' : 'Create new team'}>
            <Row>
                <Col span={18} offset={6}>
                    <div className="col-12 col-offset-6">
                        <Form layout="vertical" className="team-form" onSubmit={handleSubmit}>
                            <Form.Item label="Product owner" {...formLayout}>
                                {getFieldDecorator('productOwner', {
                                    rules: rules.productOwner,
                                    initialValue: (team && team.productOwner && team.productOwner.id) || undefined
                                })(
                                    <SearchableSelectPresenter
                                        data={productOwnerOptions}
                                        mode="single"
                                        loading={loadingProductOwner}
                                        onSearchCallback={setSearchProductOwnerInput}
                                    />
                                )}
                            </Form.Item>
                            <Form.Item label="Team Members" {...formLayout}>
                                {getFieldDecorator('teamMembers', {
                                    rules: rules.teamMembers,
                                    initialValue:
                                        (team && team.teamMembers && team.teamMembers.map(item => item && item.id)) ||
                                        undefined
                                })(
                                    <SearchableSelectPresenter
                                        data={teamMembersOptions}
                                        mode="multiple"
                                        loading={loadingTeamMembers}
                                        onSearchCallback={setSearchTeamMembersInput}
                                    />
                                )}
                            </Form.Item>
                            <Form.Item {...buttonItemLayout} className="form-buttons-wrapper">
                                <Button type="default" className="margin-right-10">
                                    Cancel
                                </Button>
                                <Button type="default" icon="user" htmlType="submit" className="submit-button">
                                    {team ? 'Update team' : 'Create team'}
                                </Button>
                            </Form.Item>
                        </Form>
                    </div>
                </Col>
            </Row>
        </Card>
    );
};
