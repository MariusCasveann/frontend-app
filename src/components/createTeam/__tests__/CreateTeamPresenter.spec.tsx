import React from 'react';
import renderer from 'react-test-renderer';

jest.doMock('antd', () => ({
    Button: 'Button',
    Col: 'Col',
    Form: 'Form',
    Row: 'Row'
}));

jest.mock('../../common/molecules/SearchableSelect/SearchableSelectPresenter', () => 'SearchableSelectPresenter');

import CreateTeamPresenter from '../CreateTeamPresenter';
import { mockEmptyTeam, mockForm, mockFunction } from '../../common/mock/MockData';

describe('CreateTeamPresenter', () => {
    it('renders a card with the team needed fields', () => {
        const mockData = [
            {
                id: 351952,
                label: 'Test test'
            },
            {
                id: 80503,
                label: 'Test1 test1'
            },
            {
                id: 412605,
                label: 'Test2 test2'
            }
        ];
        const wrapper = renderer
            .create(
                <CreateTeamPresenter
                    handleSubmit={mockFunction}
                    teamMembersOptions={mockData}
                    productOwnerOptions={mockData}
                    setSearchProductOwnerInput={mockFunction}
                    setSearchTeamMembersInput={mockFunction}
                    form={mockForm}
                    loadingTeamMembers={false}
                    loadingProductOwner={false}
                    team={mockEmptyTeam}
                />
            )
            .toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});
