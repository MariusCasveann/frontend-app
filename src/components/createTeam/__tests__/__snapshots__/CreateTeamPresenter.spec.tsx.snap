// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTeamPresenter renders a card with the team needed fields 1`] = `
<div
  className="card"
  style={
    Object {
      "flex": "1 1 100%",
    }
  }
>
  <div
    className="card-title-wrapper"
  >
    <div
      className="card-title"
    >
      Update Team
    </div>
  </div>
  <div
    className="card-content-data"
  >
    <div
      className="card-content-wrapper"
    >
      <div
        className="ant-row"
      >
        <div
          className="ant-col ant-col-18 ant-col-offset-6"
        >
          <div
            className="col-12 col-offset-6"
          >
            <form
              className="ant-form ant-form-vertical team-form"
              onSubmit={[MockFunction]}
            >
              <div
                className="ant-row ant-form-item"
              >
                <div
                  className="ant-col ant-col-14 ant-form-item-label"
                >
                  <label
                    className=""
                    htmlFor={null}
                    onClick={[Function]}
                    title="Product owner"
                  >
                    Product owner
                  </label>
                </div>
                <div
                  className="ant-col ant-col-14 ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    />
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
              >
                <div
                  className="ant-col ant-col-14 ant-form-item-label"
                >
                  <label
                    className=""
                    htmlFor={null}
                    onClick={[Function]}
                    title="Team Members"
                  >
                    Team Members
                  </label>
                </div>
                <div
                  className="ant-col ant-col-14 ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    />
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item form-buttons-wrapper"
              >
                <div
                  className="ant-col ant-col-14 ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    >
                      <button
                        className="ant-btn margin-right-10 ant-btn-default"
                        onClick={[Function]}
                        type="button"
                      >
                        <span>
                          Cancel
                        </span>
                      </button>
                      <button
                        className="ant-btn submit-button ant-btn-default"
                        onClick={[Function]}
                        type="submit"
                      >
                        <i
                          aria-label="icon: user"
                          className="anticon anticon-user"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="user"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                            />
                          </svg>
                        </i>
                        <span>
                          Update team
                        </span>
                      </button>
                    </span>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
